app/routes/client.py

```diff
@@ -216,38 +229,42 @@ def event(slug):
     :param str slug: The unique slug ID for the post.
     """
     if Event.objects(published=True, slug=slug).count() == 0:
-        abort(404) # Either invalid event ID or duplicate IDs.
+        abort(404)  # Either invalid event ID or duplicate IDs.
 
     event = Event.objects(published=True, slug=slug)[0]
 
-
     if event.is_recurring:
-        upcoming_event_instances = Event.objects(published=True,
-                                                 start_date__gte=date.today(),
-                                                 slug=slug).order_by('start_date')
+        upcoming_event_instances = (Event.objects(published=True,
+                                                  start_date__gte=date.today(),
+                                                  slug=slug)
+                                         .order_by('start_date'))
         if upcoming_event_instances:
             event = upcoming_event_instances[0]
         else:
             event = event.parent_series.events[-1]
 
-        upcoming_events = Event.objects(published=True,
-                                        start_date__gte=date.today(),
-                                        id__ne=event.id).order_by('start_date')[:3]
+        upcoming_events = (Event.objects(published=True,
+                                         start_date__gte=date.today(),
+                                         id__ne=event.id)
+                                .order_by('start_date')
+                                .limit(3))
 
 
         return render_template('events/event.html',
                                event=event,
                                upcoming_events=upcoming_events)
 
-    upcoming_events = Event.objects(published=True,
-                                    start_date__gte=date.today(),
-                                    id__ne=event.id).order_by('start_date')[:3]
+    upcoming_events = (Event.objects(published=True,
```

Comment:
this query is identical to one happening on 246-250. I suspect there's some context I'm missing but it seems like you could generalize this.
I also see this query happening other places. Usually with the same magic number of 3